### DISCUSSION TITLE
[Design][Microservices] Do we really need the <Preserve files> check on configmaps?

### DISCUSSION THREAD

**Discussion by frengooo** [mia-platform/community#225] [Q&A] [answered]
### Which section?

Console

### Which is the version?

v10.7.0

### Description

Hi guys,
I just spent a lot of time on a problem, without understanding the reason.
After a couple of hours, a colleague noticed that a file that our microservice needs to properly work was missing because of this check:
<img width="579" alt="Screenshot 2023-03-31 at 12 48 27" src="https://user-images.githubusercontent.com/28152586/229100443-cda2eb74-2c49-4fa4-ad88-ea06f8380085.png">

Do you think we really need this check? Why should we remove all files in a specific mount path when adding our config map?

If we need, can you check the flag by default? It was very very difficult to find the problem.

**Comment by fredmaggiowski** [comment]
Hi, I'm sorry you had an hard time, however removing files from an existing directory when mounting a ConfigMap (or Secret) is the default behaviour of kubernetes; there are times when you would like to add or overwrite specific files from the directory and for such use cases the flag is useful allowing the user to decide how to treat other files that may exist in the same mount path

**Reply by frengooo** [reply]
Hi @fredmaggiowski, what do you think about enabling that check by default? I think that an eventual bug could be easy to be found in this way.

**Reply by GiovannaMonti** [reply]
When you first create a microservice, you can set the `usePreserve` boolean to check the flag by default. There is a reference to this flag in the Marketplace docs
https://docs.mia-platform.eu/docs/marketplace/add_to_marketplace/contributing_overview#common-to-microservice-components-plugins-templates-examples

**Reply by frengooo** [reply]
Hi @GiovannaMonti, thanks, I didn't know that. But this is valid for default config-maps of marketplace components, not for new config-map or custom microservices.

**Reply by GiovannaMonti** [reply]
Hi, sorry, yes I was afraid that it would not suit your situation, but I thought it might still be helpful to know about the existence of the flag. Maybe we could extend it

**Reply by fredmaggiowski** [reply, marked as answer]
> what do you think about enabling that check by default

@silversoul93 this could be interesting, however it would be a breaking change on how ConfigMaps/Secrets are managed, not only in terms of user expectations inside the Console but also for user expectations on how Kubernetes works. For this reason I think is something that we can consider but I'm not sure is the right thing to do 🤔
